compiler/rustc_next_trait_solver/src/solve/mod.rs

```diff
@@ -148,6 +148,37 @@ where
         }
     }
 
+    fn compute_unstable_feature_goal(
+        &mut self,
+        param_env: <I as Interner>::ParamEnv,
+        symbol: <I as Interner>::Symbol,
+    ) -> QueryResult<I> {
+        // Iterate through all goals in param_env to find the one that has the same symbol.
+        for pred in param_env.caller_bounds().iter() {
+            if let ty::ClauseKind::UnstableFeature(sym) = pred.kind().skip_binder() {
+                if sym == symbol {
+                    return self.evaluate_added_goals_and_make_canonical_response(Certainty::Yes);
+                }
+            }
+        }
+
+        // During codegen we must assume that all feature bounds hold as we may be
+        // monomorphizing a body from an upstream crate which had an unstable feature
+        // enabled that we do not.
+        //
+        // Note: `feature_bound_holds_in_crate` does not consider a feature to be enabled
+        // if we are in std/core even if there is a corresponding `feature` attribute on the crate.
+        if self.cx().features().feature_bound_holds_in_crate(symbol)
+            || (self.typing_mode() == TypingMode::PostAnalysis)
```

Comment:
pls add comment why `Ambiguity` instead of error, also, flip these conditions:

```suggestion
        if (self.typing_mode() == TypingMode::PostAnalysis) ||
            self.cx().features().feature_bound_holds_in_crate(symbol)
```